OAuth2/Client/Impl/GitHubClient.cs

```diff
@@ -87,5 +124,17 @@ protected override Endpoint UserInfoServiceEndpoint
         {
             get { return new Endpoint { BaseUri = "https://api.github.com/", Resource = "/user" }; }
         }
+
+        protected virtual Endpoint UserEmailServiceEndpoint
+        {
+            get { return new Endpoint { BaseUri = "https://api.github.com/", Resource = "/user/emails" }; }
+        }
+
+        protected class UserEmails
+        {
+            public string Email { get; set; }
+            public bool Primary { get; set; }
```

Comment:
It would be better to have it named as `IsPrimary`, according to MS naming conventions.